libethereum/FastSync.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "FastSync.h"
+#include "EthereumPeer.h"
+
+#include <libethcore/BlockHeader.h>
+
+using namespace std;
+using namespace dev;
+using namespace dev::eth;
+
+namespace
+{
+unsigned const c_maxRequestHeaders = 1024;
+}
+
+FastSync::FastSync()
+{
+}
+
+void FastSync::onPeerStatus(shared_ptr<EthereumPeerFace> _peer)
```

Comment:
FastSync saves them only as weak_ptr's now.
BlockChainSync does the same, I don't see yet where it takes ownership...